pkgs/native_assets_builder/test_data/complex_link/hook/build.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+
+const packageName = 'complex_link';
+
+void main(List<String> args) async => build(args, (config, output) async {
+      output.addAssets(
```

Comment:
Nit: Prefer list literals with inline ifs/loops

```dart
      output.addAssets(
        [
          for (int index = 0; index < 2; index++)
            DataAsset(
              name: 'data_$index',
              // TODO(mosuem): Simplify specifying files/file paths
              file: config.packageRoot.resolve('data_$index.json'),
              package: packageName,
            ),
        ],
        linkInPackage: 'complex_link',
      );
```

(This prevents having a lambda and the less commonly used `List.generate`.)

Reply:
I think I have to disagree with Lasse here. The style guide also does not address this, so I would rather keep it as is.

Reply:
I have used list literals everywhere else in this repository, so now we're introducing inconsistent style. I don't care so much about the style of the implementation, but the style of our doc comments and examples that users might copy, I strongly prefer to be consistent.

Reply:
Well this is just test data ;) The examples and docs don't use `List.generate`.

Reply:
I might have the urge to refactor it if I ever touch this code. 😆

Reply:
Please refactor to your liking :) Maybe `List.generate` will grow on you until then..